src/auth/auth.gql-guard.ts

```diff
@@ -0,0 +1,30 @@
+import { Injectable, CanActivate, ExecutionContext } from '@nestjs/common';
+import { GqlExecutionContext } from '@nestjs/graphql';
+import { Observable } from 'rxjs';
+import { FirebaseService } from '../firebase';
+
+@Injectable()
+export class GqlAuthGuard implements CanActivate {
+  constructor(private readonly firebaseService: FirebaseService) {}
+
+  canActivate(
+    context: ExecutionContext,
+  ): boolean | Promise<boolean> | Observable<boolean> {
```

Comment:
Shouldn't it be just Promise<boolean> | Observable<boolean> ?

Reply:
This is a need because of the CanActivate interface definitions.
```
/**
 * Interface defining the `canActivate()` function that must be implemented
 * by a guard.  Return value indicates whether or not the current request is
 * allowed to proceed.  Return can be either synchronous (`boolean`)
 * or asynchronous (`Promise` or `Observable`).
 *
 * @see [Guards](https://docs.nestjs.com/guards)
 *
 * @publicApi
 */
```